enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,235 @@
+
+60 minutes?  That seems high!  That's because it is high.  We want to reduce
+the noise.  We've done a lot to make clusters and operators auto-heal
+themselves.  The whole idea is that if a condition has persisted for more than
+60 minutes, it's unlikely to be resolved without intervention any time later.
+
+#### Example 1: All machine-config-daemons are crashlooping.
+
+That seems significant
+and it probably is, but nobody has to get out of bed to fix this **right now**.
+But, what if a machine dies, I won't be able to get a replacement?  Yeah, that
+is probably true, but how often does that happen?  Also, that's an entirely
+separate set of concerns.
+
+#### Example 2:  I have an operator that needs at least 3 of something to work.
+* With 2/3 replicas, warning alert after 60M.
+* With 1/3 replicas, warning alert after 10M.
+* With 0/3 replicas, warning alert after 5M.
+
+Q: How long should we wait until the above conditions rises to a 'critical'
+alert?
+
+A: It should never rise to the level of a critical alert, it's not critical.  If
+it was, this section would not apply.
+
+#### Example 3:  I have an operator that only needs 1 replica to function
+
+If the cluster can upgrade with only 1 replica, and the the service is
+available despite other replicas being unavailable, this can probably
+be just an info-level alert.
+
+#### Q: What if a particular condition will block upgrades?
+
+A: It's a warning level alert.
+
+
+### Alert Ownership
+
+Previously, the bulk of our alerting was handled directly by the monitoring
+team.  Initially, this gave use some indication into platform status without
+much effort by each component team.  As we mature our alerting system, we should
+ensure all teams take ownership of alerts for their individual components.
+
+Going forward, teams will be expected to own alerting rules within their own
+repositories.  This will reduce the burden on the monitoring team and better
+enable component owners to control their alerts.
+
+Additional responsibilities include:
+
+1. First in line to receive the bug for an alert firing
+1. Responsible for describing, "what does it mean" and "what does it do"
+1. Responsible for choosing the level of alert
+1. Responsible for deciding if the alert even matters
+
+
+### Documentation Required
+
+1. The name of the alerting rule should clearly identify the component impacted
+by the issue (for example etcdInsufficientMembers instead of
+InsufficientMembers, MachineConfigDaemonDrainError instead of MCDDrainError).
+It should camel case, without whitespace, starting with a capital letter. The
+first part of the alert name should be the same for all alerts originating from
+the same component.
+1. Alerting rules should have a "severity" label whose value is either info,
+warning or critical (matching what we have today and staying aside from the
+discussion whether we want minor or not).
+1. Alerting rules should have a description annotation providing details about
```

Comment:
I would like to see a follow up to this that adds examples that people can prescriptively follow, maybe with 1-3 cherry-picked examples of "good descriptions" including some guidance.  Or we can put htat in a "how to write alerts doc" and link here.